.github/workflows/monitor_self_hosted_runners.yml

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Monitor Self-Hosted Runners Status
+on:
+  schedule:
+    - cron: "0 */12 * * *"
+  workflow_dispatch:
+jobs:
+  monitor-runners:
+    name: Monitor Self-hosted Runners Status
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0
+    - name: Set up node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    - name: Setup checkout
+      uses: actions/checkout@v3
+    - name: Setup GCP account
+      run: |
+            echo "${{ secrets.GCP_PLAYGROUND_SA_KEY }}" | base64 -d > /tmp/gcp_access.json
+            which gcloud
+            gcloud auth activate-service-account --project=apache-beam-testing --key-file=/tmp/gcp_access.json
+    - name: Exporting ID
+      run: echo "IDENTITY_TOKEN=$(gcloud auth print-identity-token)" >> $GITHUB_ENV
+            
+    - name: Run monitor script
+      run: |
+        npm install
+        node sendRunnersReport.js
+      working-directory: 'scripts/ci/self-hosted-runners-report' 
+      env:
+        ISSUE_REPORT_SENDER_EMAIL_ADDRESS: "${{ secrets.ISSUE_REPORT_SENDER_EMAIL_ADDRESS }}"
+        ISSUE_REPORT_SENDER_EMAIL_PASSWORD: ${{ secrets.ISSUE_REPORT_SENDER_EMAIL_PASSWORD }}
+        ISSUE_REPORT_RECIPIENT_EMAIL_ADDRESS: "dev@beam.apache.org"
+        ISSUE_REPORT_SENDER_EMAIL_SERVICE: "gmail"
+        ENDPOINT: "https://us-central1-apache-beam-testing.cloudfunctions.net/monitorRunnersStatus" #we suggest adding this ENDPOINT as a repo secret too
```

Comment:
> we suggest adding this ENDPOINT as a repo secret too

Does keeping this secret actually get us anything?

Reply:
The Cloud Function requires authentication for invocation and it's configured to only allow calls within the same VPC so keeping it public shouldn't affect security.

Reply:
Cool, lets keep it public then and remove the comment